docs/guides/initializers.md

```diff
@@ -0,0 +1,123 @@
+
++++ {"id": "6nsUREar3k1H"}
+
+`Initializers` are functions that can be passed as optional arguments to the kernel initializer (`kernel_init`) and the bias initializer (`bias_init`) if you want to specify how the parameters of a Module layer are initialized. A full list of Flax initializers can be found [here](https://flax.readthedocs.io/en/latest/api_reference/flax.linen.html#module-flax.linen.initializers), and are in fact, the same as the [JAX initializers](https://jax.readthedocs.io/en/latest/jax.nn.initializers.html).
+
+The default kernel initializer is [`flax.linen.initializers.lecun_normal`](https://flax.readthedocs.io/en/latest/api_reference/_autosummary/flax.linen.initializers.lecun_normal.html) and the default bias initializer is [`flax.linen.initializers.zeros`](https://flax.readthedocs.io/en/latest/api_reference/_autosummary/flax.linen.initializers.zeros.html).
+
+```{code-cell}
+---
+executionInfo:
+  elapsed: 118
+  status: ok
+  timestamp: 1673133040623
+  user:
+    displayName: Marcus Chiam
+    userId: '17531616275590396120'
+  user_tz: 480
+id: 8Gvqoi_rzII5
+---
+import jax, jax.numpy as jnp
+import flax.linen as nn
+from flax.linen.initializers import lecun_normal, zeros
+```
+
+```{code-cell}
+---
+executionInfo:
+  elapsed: 61
+  status: ok
+  timestamp: 1673133040851
+  user:
+    displayName: Marcus Chiam
+    userId: '17531616275590396120'
+  user_tz: 480
+id: nHyw_azR0TAL
+---
+layer = nn.Dense(features=5)
+default_params = layer.init(jax.random.PRNGKey(42), jnp.empty((1,4)))['params']
+
+layer = nn.Dense(features=5, kernel_init=lecun_normal(), bias_init=zeros)
+manual_params = layer.init(jax.random.PRNGKey(42), jnp.empty((1,4)))['params']
+
+# should get the same param values
+assert (default_params['kernel'] == manual_params['kernel']).all()
+assert (default_params['bias'] == manual_params['bias']).all()
+```
+
++++ {"id": "jPW_E1ZqHdrQ"}
+
+## `Initializer` function signature
+
++++ {"id": "3Kglqd9vuxTG"}
+
+To maintain consistency, all `Initializer` functions that are passed to the `kernel_init` and `bias_init` arguments **must follow the function signature: `[PRNGKey, Shape, Dtype] -> Array`**. Most functions in the [Flax initializer list](https://flax.readthedocs.io/en/latest/api_reference/flax.linen.html#module-flax.linen.initializers) are **builder functions** and build an `Initializer` function that follows this function signature. The two exceptions are [`flax.linen.initializers.zeros`](https://flax.readthedocs.io/en/latest/api_reference/_autosummary/flax.linen.initializers.zeros.html) and [`flax.linen.initializers.ones`](https://flax.readthedocs.io/en/latest/api_reference/_autosummary/flax.linen.initializers.ones.html), which are already `Initializer` functions that follow the function signature. This is why in the above example, we must call `lecun_normal()` to build an `Initializer` function, whereas we can directly use `zeros` since it's already an `Initializer` function.
```

Comment:
Thinking of "prefer simple APIs over more documentation", I am wondering whether it wouldn't be easier to just add two fake builder functions for `zeros` and `ones` as follows:

```py
def zeros(dtype: DTypeLikeInexact = jnp.float_) -> Array:
  def init(key: KeyArray,
           shape: core.Shape,
           dtype: DTypeLikeInexact = dtype) -> Array:
    del key
    return jnp.zeros(shape, dtype)
```

Then we don't need this complicated explanation and people can simply use all initializers consistently. WDYT?

Reply:
I think this is a great idea! It would make things much more consistent. The question is what we should do with the original [flax.linen.initializers.zeros](https://flax.readthedocs.io/en/latest/api_reference/_autosummary/flax.linen.initializers.zeros.html) and [flax.linen.initializers.zeros](https://flax.readthedocs.io/en/latest/api_reference/_autosummary/flax.linen.initializers.ones.html). If we replace them with the fake builder functions you suggested, then wouldn't this break code using these initializers? On the other hand if we leave them, then I think users may get confused on what the difference is; i.e. which they should use, which ones to call versus which ones to use explicitly, etc.